packages/navigator/src/browser/navigator-widget.ts

```diff
@@ -61,6 +63,25 @@ export class FileNavigatorWidget extends FileTreeWidget {
         super.onAfterAttach(msg);
         this.addClipboardListener(this.node, 'copy', e => this.handleCopy(e));
         this.addClipboardListener(this.node, 'paste', e => this.handlePaste(e));
+        // Add DragEvent to prevent the user to drop a file in the navigator area
```

Comment:
Why not do it in `FileTreeWidget`?

Reply:
The DragEvent are in the navigator-widget to prevent the end-user to drop a file anywhere in the navigator. It would result in leaving THEIA and open the file in the browser which is not what the end-user would expect here. So whenever a user DRAG a file/folder over the THEIA navigator, he will remain in THEIA and he by accident he drops the file/folder over an empty space or over a file, the DROP will not copy the file, but at least the user will remain in THEIA. When selecting a proper dropzone in the navigator (folder), the css file will indicates it is a valid destination (changing the color) and when DROPPING, the files will be copied in that folder

Reply:
Why they cannot be installed in `FileTreeWidget` and call `handleDragEnterEvent` with `this.model.root`? The same for other events. I would like to have the logic drag & drop encapsulated in one place, which can be overridden if it is required by subclasses.

Reply:
Done in FileTreeWidget, not anymore in the navigator-widget